Fw/DataStructures/RedBlackTreeSetOrMapImpl.hpp

```diff
@@ -0,0 +1,1699 @@
+                //                    B   K4 (sibling)  B         | black height i + 1 |
+                //                    B                 B         |                    |
+                //                     BBBBBBBBBBBBBBBBB          ----------------------
+                //                         /       \
+                //                        /         \
+                //                       V           V
+                //     RRRRRRRRRRRRRRRRRRRR       ------------------------
+                //    R                    R      |                      |
+                //    R K2 (distantNephew) R      |  black height i + 1  |
+                //    R                    R      |                      |
+                //     RRRRRRRRRRRRRRRRRRRR       ------------------------
+                //           |      |
+                //           |      |
+                //           V      V
+                //   ------------------------
+                //   |                      |
+                //   |  black height i + 1  |
+                //   |                      |
+                //   ------------------------
+                */
+                this->removeBlackLeafNodeHelper2(parent, sibling, distantNephew, direction);
+                // The subtree has this shape. The entire tree is a valid red-black tree.
+                /*
+                //                     BBBBBBBBBBBBBBBBB
+                //                    B                 B
+                //                    B   K4 (sibling)  B
+                //                    B                 B
+                //                     BBBBBBBBBBBBBBBBB
+                //                         /       \
+                //                        /         \
+                //                       V           V
+                //     BBBBBBBBBBBBBBBBBBBB         BBBBBBBBBBBBBBBBB
+                //    B                    B       B                 B
+                //    B K2 (distantNephew) B       B   K6 (parent)   B
+                //    B                    B       B                 B
+                //     RRRRRRRRRRRRRRRRRRRR         BBBBBBBBBBBBBBBBB
+                //           |      |                    |     |
+                //           |      |                    |     |
+                //           V      V                    V     V
+                //   ------------------------   ------------------------
+                //   |                      |   |                      |
+                //   |  black height i + 1  |   |  black height i + 1  |
+                //   |                      |   |                      |
+                //   ------------------------   ------------------------
+                */
+                done = true;
+            } else if (this->getNodeColor(parent) == Color::RED) {
+                this->m_nodes[sibling].m_color = Color::RED;
```

Comment:
Missed the start diagram in this case.

Reply:
OK. It's the same as the one below, with two different node colors. I thought it wasn't necessary to replicate the diagram.